crates/storage/trie/trie.rs

```diff
@@ -182,15 +183,15 @@ impl Trie {
             .unwrap_or(*EMPTY_TRIE_HASH)
     }
 
-    #[cfg(test)]
+    #[cfg(all(test, feature = "libmdbx"))]
     /// Creates a new Trie based on a temporary Libmdbx DB
     fn new_temp() -> Self {
         let db = test_utils::new_db::<test_utils::TestNodes>();
         Trie::new(Box::new(LibmdbxTrieDB::<test_utils::TestNodes>::new(db)))
     }
```

Comment:
Could we modify this to return either a Libmdbx or an InMemory based trie depending on features? I would like to avoid disabling the whole test suite if a feature is not active

Reply:
Sure, working on it.

Reply:
I've just finished it.